---
title: C28105 |Microsoft 文件
ms.custom: ''
ms.date: 11/04/2016
ms.reviewer: ''
ms.suite: ''
ms.technology:
- vs-ide-code-analysis
ms.tgt_pltfrm: ''
ms.topic: article
f1_keywords:
- C28105
helpviewer_keywords:
- C28105
ms.assetid: 5afdec78-30eb-470c-980f-93d0cb3e74cc
caps.latest.revision: 3
author: mikeblome
ms.author: mblome
manager: ghogen
ms.workload:
- multiple
ms.openlocfilehash: 21b1f4923621105f84ea8dca78fb31006fffde49
ms.sourcegitcommit: 32f1a690fc445f9586d53698fc82c7debd784eeb
ms.translationtype: MT
ms.contentlocale: zh-TW
ms.lasthandoff: 12/22/2017
---
# <a name="c28105"></a>C28105
警告 C28105： 洩漏資源，因為發生例外狀況  
  
 指定的資源不會釋放時引發的例外狀況。 路徑所指定的陳述式可以引發例外狀況。 這項警告是類似於警告[C28103](../code-quality/c28103.md)，不過，在此情況下相關例外狀況。  
  
## <a name="example"></a>範例  
 下列程式碼範例會產生此警告：  
  
```  
res = KeSaveFloatingPointState(buffer);  
  
res = AllocateResource(Resource);  
char *p2 = new char[10]; // could throw  
  
delete[] p2;  
FreeResource(Resource)  
```  
  
 下列程式碼範例可避免此警告：  
  
```  
res = AllocateResource(Resource);  
char *p2;  
  
try {  
    p2 = new char[10];  
} catch (std::bad_alloc *e) {  
    // just handle the throw  
    ;  
}  
FreeResource(Resource)  
```